bech32-migration/auth/auth.go

```diff
@@ -0,0 +1,44 @@
+package auth
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+
+	"github.com/likecoin/likechain/bech32-migration/utils"
+)
+
+func MigrateAddressBech32(ctx sdk.Context, storeKey sdk.StoreKey, cdc codec.BinaryCodec) {
+	ctx.Logger().Info("Migration of address bech32 for auth module begin")
+	accountCount := uint64(0)
+	utils.IterateStoreByPrefix(ctx, storeKey, types.AddressStoreKeyPrefix, func(bz []byte) []byte {
+		var accountI types.AccountI
+		err := cdc.UnmarshalInterface(bz, &accountI)
+		if err != nil {
+			panic(err)
+		}
+		switch accountI.(type) {
+		case *types.BaseAccount:
+			{
+				acc := accountI.(*types.BaseAccount)
+				acc.Address = utils.ConvertAccAddr(acc.Address)
+			}
+		case *types.ModuleAccount:
+			{
+				acc := accountI.(*types.ModuleAccount)
+				acc.Address = utils.ConvertAccAddr(acc.Address)
+			}
+		}
```

Comment:
Can we catch unhandled type here and panic if occurs?

Reply:
I think we should not panic, since un-migrated accounts still work.
I think it's more appropriate to add a log for warning here.
Did some code search and found that the only account types in Cosmos SDK that we are not migrating are vesting accounts.
Originally I thought we don't have vesting accounts on LikeCoin chain, but it seems possible to create one with transaction, so I will add vesting accounts for migration.

Reply:
i don't think we have vesting module integrated, but good to catch all possible cases here